lib/hasPermissions.ts

```diff
@@ -233,8 +210,16 @@ export class HasPermissionsDirectiveVisitor<
         were executed before it, then pass or extend that array with the new
         permissions
       */
-      const existingMissingPermissions = args[missingPermissionsArgumentName];
+      const existingMissingPermissions = ((info as unknown) as Record<
```

Comment:
do you know if `if (missingPermissionsArgumentName in info)` fixes it? Or maybe try with the explicit literal name (you can remove the `missingPermissionsArgumentName`).